colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +211,71 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+"""
+Global variable for the cached CMake version number.
+
+When valid, this will be a pkg_resources.extern.packaging.version.Version.
+It may also be None when the CMake version could not be determined to avoid
+trying to determine it again.
+"""
+_cached_cmake_version = False
+
+
+def get_cmake_version():
+    """
+    Get the CMake version.
+
+    The function caches the result on the first invocation and reuses that on
+    subsequent invocations.
+
+    :returns: The version as reported by `CMAKE_EXECUTABLE --version`, or None
+      when the version number could not be determined
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    global _cached_cmake_version
+    if _cached_cmake_version is False:
+        _cached_cmake_version = _parse_cmake_version()
+    return _cached_cmake_version
+
+
+def parse_cmake_version_string(version_string):
+    """
+    Parse the given CMake version string.
+
+    This is for internal use and testing.
```

Comment:
Since the function is meant to be internal I removed this line from the docblock and instead prefixed the function name with an underscore: 631e008b53811e6031dc42cb30d3d1b6db27cf15.